### DISCUSSION TITLE
SSL Update

### DISCUSSION THREAD

**Discussion by gprao08** [Sunbird-cQube/community#831] [General]
Hi Team
Could you plz help me to update SSL to my portal as the old SSL expired.

**Comment by RNarmatha** [comment]
@bommalla - please support them for updating the SSL certificate

**Comment by bommalla** [comment]
@gprao08 Please follow the below steps to updated the SSl certificate:

Step1 : Please move the validate SSL keys to cQube home directory.
Step2 : Please move SSL keys from cQube home directory to below mentioned path
        sudo docker exec -it nginx_app bash
        cd /etc/ssl/vsk.ap.gov.in
Step3 : After the please navigate the below mentioned path and modify the nginx template
        cd cqube-devops
        cd /opt/cqube/nginx
        vim default.conf
Step4: Please go inside default.conf and modify the below mentioned two lines.

       ssl_certificate /etc/ssl/vskndear.sikkim.gov.in/sslcert.crt;
       ssl_certificate_key /etc/ssl/vskndear.sikkim.gov.in/sslcert.key;
Step5: After modifying the SSL path then we can restart the nginx container
       sudo docker restart nginx_app

**Reply by gprao08** [reply]
@bommalla 
After I gave 
sudo docker exec -it nginx_app bash
cd /etc/ssl/vsk.ap.gov.in
the files has not moved to the above location. Screen shots attached for your reference. The new files are wildap1.crt & star1.key. Could you plz help us.
![screen1](https://github.com/Sunbird-cQube/community/assets/143190553/672d0ac7-e2f4-483b-9443-a709879d108e)
![screen2](https://github.com/Sunbird-cQube/community/assets/143190553/d6de9070-4bbb-4c3a-97c4-1e156d9cf7f7)

**Comment by bommalla** [comment]
@gprao08 Please use the below copy command to move the SSL files.

sudo docker cp <SSL file name> nginx_app:/etc/ssl/vsk.ap.gov.in .

**Reply by gprao08** [reply]
@bommalla @RNarmatha 
could you plz provide the command with proper syntax.. The original file Source is /home/SSL/wildap1.crt and /home/SSL/star1.key and the destination is nginx_app:/etc/ssl/vsk.ap.gov.in
When I apply the command as above after I go to the source directory /home/SSL it is not copying files and shows error. Screen shot attached for reference.
![screen3](https://github.com/Sunbird-cQube/community/assets/143190553/4c87c5f7-b3fe-4060-b91d-10c362c7f670)

**Comment by bommalla** [comment]
@gprao08 Please find the below command.

sudo docker cp wildap1.crt star1.key nginx_app:/etc/ssl/vsk.ap.gov.in .

**Reply by gprao08** [reply]
@bommalla Tq now it is working. I just changed the above command to sudo docker cp . nginx_app:/etc/ssl/vsk.ap.gov.in
